ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -41,6 +41,41 @@ EOF
     fi
 }
 
+function update_cloud_credentials(){
+    platform=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')
+    cco_mode=$(oc get cloudcredential cluster -o jsonpath='{.spec.credentialsMode}')
+    if (( SOURCE_MINOR_VERSION == TARGET_MINOR_VERSION )) || (( SOURCE_MINOR_VERSION < 8 )) || [[ "${cco_mode}" != "Manual" ]]; then
+        echo "No need update cloud credentials" && return
+    fi
+    preCredsDir="/tmp/pre-include-creds"
+    tobeCredsDir="/tmp/tobe-include-creds"
+    mkdir "${preCredsDir}" "${tobeCredsDir}"
+    # Extract all CRs from live cluster with --included
+    if ! oc adm release extract --to "${preCredsDir}" --included --credentials-requests; then
+        echo "Failed to extract CRs from live cluster!" && exit 1
+    fi
+    if ! oc adm release extract --to "${tobeCredsDir}" --included --credentials-requests "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"; then
+        echo "Failed to extract CRs from tobe upgrade release payload!" && exit 1
+    fi
+
+    # TODO: add gcp and azure
+    # Update iam role with ccoctl based on tobeCredsDir
+    if [[ "${platform}" == "AWS" ]]; then
+        if [[ ! -e ${SHARED_DIR}/aws_oidc_provider_arn ]]; then
+            echo "No aws_oidc_provider_arn file in SHARED_DIR" && exit 1
+        else
+            oidc_provider=$(head -n1 ${SHARED_DIR}/aws_oidc_provider_arn)
+        fi
+    fi
+    if ! diff -r "${preCredsDir}" "${tobeCredsDir}" &> /dev/null; then
+        infra_name=${NAMESPACE}-${UNIQUE_HASH}
+        if ! ccoctl aws create-iam-roles --name="${infra_name}" --region="${LEASED_RESOURCE}" --credentials-requests-dir="${tobeCredsDir}" --identity-provider-arn="${oidc_provider}"; then
```

Comment:
I think this command should run in the if block on line 63.

Reply:
good catch, updated